RepositoryCompiler/CodeModel/CodeParsers/CSharp/CSharpCodeParser.cs

```diff
@@ -199,6 +204,78 @@ private CaDETClass LinkOuterClass(List<CaDETClass> classes, string containerName
         {
             return classes.FirstOrDefault(c => c.FullName.Equals(containerName));
         }
+        
+        private List<CaDETClass> LinkFieldTypes(List<CaDETClass> classes, List<CaDETField> fields)
+        {
+            List<CaDETClass> fieldTypes = new List<CaDETClass>();
+            foreach (var f in fields)
+            {
+                fieldTypes.AddRange(FindTypes(classes, new List<CaDETClass>() { f.Type }));
+            }
+            return fieldTypes.Where(t => t != null).Distinct().ToList();
+        }
+        private List<CaDETClass> LinkMethodReturnTypes(List<CaDETClass> classes, List<CaDETMember> members)
+        {
+            var methods = members.Where(m => m.Type == CaDETMemberType.Method).ToList();
+            List<CaDETClass> returnTypes = new List<CaDETClass>();
+            foreach (var m in methods)
+            {
+                returnTypes.AddRange(FindTypes(classes, new List<CaDETClass>() { m.ReturnType}));
+            }
+            return returnTypes.Where(t => t != null).Distinct().ToList();
+        }
+
+        private List<CaDETClass> LinkMethodVariableTypes(List<CaDETClass> classes, List<CaDETMember> members)
+        {
+            var methods = members.Where(m => m.Type == CaDETMemberType.Method).ToList();
+            List<CaDETClass> variableTypes = new List<CaDETClass>();
+            foreach (var m in methods)
+            {
+                variableTypes.AddRange(FindTypes(classes, m.VariableTypes));
+            }
+            return variableTypes.Where(t => t != null).Distinct().ToList();
+        }
+
+        private List<CaDETClass> FindTypes(List<CaDETClass> classes, List<CaDETClass> types)
+        {
+            List<CaDETClass> foundTypes = new List<CaDETClass>();
+            foreach (var type in types)
+            {
+                foundTypes.Add(classes.FirstOrDefault(c => c.Name.Equals(type.Name)));
```

Comment:
This foreach will add a bunch of nulls, which is why you need to repeat the Where(t => t != null) in all of the above methods instead of doing it once at the end of this method.